lib/trie/db/db.go

```diff
@@ -4,12 +4,20 @@ package db
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/ChainSafe/gossamer/lib/common"
 )
 
 type MemoryDB struct {
 	data map[common.Hash][]byte
+	l    sync.RWMutex
```

Comment:
Why is this called 1? I think I would opt for a different name unless there is good reason for it